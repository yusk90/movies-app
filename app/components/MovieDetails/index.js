import React from 'react';
import PropTypes from 'prop-types';

import './index.less';

const MovieDetails = ({
  title, rating, posterSrc, genres, year, length, description,
}) => (
  <div className="movie-details">
    <div className="movie-details__poster">
      <img className="movie-details__poster-image" src={posterSrc} alt={title} />
    </div>
    <div className="movie-details__description-wrapper">
      <h1 className="movie-details__title">
        {title} <span className="movie-details__rating">{rating}</span>
      </h1>
      <div className="movie-details__genres">
        {genres && genres.join(', ')}
      </div>
      <div className="movie-details__statistics">
        <span className="movie-details__year">{year}</span>
        <span className="movie-details__length">{length}</span>
      </div>
      <div className="movie-details__description">{description}</div>
    </div>
  </div>
);

MovieDetails.propTypes = {
  title: PropTypes.string,
  rating: PropTypes.number,
  posterSrc: PropTypes.string,
  genres: PropTypes.array,
  year: PropTypes.number,
  length: PropTypes.string,
  description: PropTypes.string,
};

export default MovieDetails;
