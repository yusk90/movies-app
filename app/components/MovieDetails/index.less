.movie-details {
  display: flex;
  font-size: 20px;

  &__poster {
    flex: 2 0 30%;
    padding: 0 50px 0 0;
  }

  &__poster-image {
    max-width: 100%;
  }

  &__description-wrapper {}

  &__title {
    display: flex;
    align-items: center;
    font-size: 40px;
    font-weight: 700;
    color: #b95866;
  }

  &__rating {
    display: inline-block;
    width: 40px;
    height: 40px;
    margin-left: 15px;
    font-size: 20px;
    color: green;
    border: 1px solid green;
    border-radius: 50%;
    line-height: 40px;
    text-align: center;
  }

  &__genre {
    margin-bottom: 20px;
  }

  &__statistics {
    margin-bottom: 20px;
    font-weight: 700;
  }

  &__year {
    margin-right: 20px;
  }

  &__length {}

  &__description {}
}
