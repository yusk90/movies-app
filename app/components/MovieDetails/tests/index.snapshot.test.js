import React from 'react';
import renderer from 'react-test-renderer';

import MovieDetails from '../index';

const movie = {
  title: 'The Godfather',
  rating: 9.2,
  posterSrc: 'https://ia.media-imdb.com/images/M/MV5BM2MyNjYxNmUtYTAwNi00MTYxLWJmNWYtYzZlODY3ZTk3OTFlXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,704,1000_AL_.jpg',
  genres: ['Crime, Drama'],
  year: 1972,
  length: '2h 55min',
  description: 'The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.',
};

describe('<MovieDetails />', () => {
  it('renders movie details', () => {
    const tree = renderer
      .create(<MovieDetails {...movie} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
