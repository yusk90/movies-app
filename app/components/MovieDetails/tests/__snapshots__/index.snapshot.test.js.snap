// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails /> renders movie details 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details__poster"
  >
    <img
      alt="The Godfather"
      className="movie-details__poster-image"
      src="https://ia.media-imdb.com/images/M/MV5BM2MyNjYxNmUtYTAwNi00MTYxLWJmNWYtYzZlODY3ZTk3OTFlXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,704,1000_AL_.jpg"
    />
  </div>
  <div
    className="movie-details__description-wrapper"
  >
    <h1
      className="movie-details__title"
    >
      The Godfather
       
      <span
        className="movie-details__rating"
      >
        9.2
      </span>
    </h1>
    <div
      className="movie-details__genres"
    >
      Crime, Drama
    </div>
    <div
      className="movie-details__statistics"
    >
      <span
        className="movie-details__year"
      >
        1972
      </span>
      <span
        className="movie-details__length"
      >
        2h 55min
      </span>
    </div>
    <div
      className="movie-details__description"
    >
      The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.
    </div>
  </div>
</div>
`;
